test/output.go

```diff
@@ -686,3 +699,230 @@ func (e *customErrorJSON) Error() string {
+`: `{"caller":"test/output.go:<LINE>","msg":"test","err":"whoops"}
+`,
+
+		`I output.go:<LINE>] "helper" akey="avalue"
+`: `{"caller":"test/output.go:<LINE>","msg":"helper","v":0,"akey":"avalue"}
+`,
+
+		`I output.go:<LINE>] "hello/world: test" akey="avalue"
+`: `{"logger":"hello.world","caller":"test/output.go:<LINE>","msg":"test","v":0,"akey":"avalue"}
+`,
+
+		`I output.go:<LINE>] "test" X="y" duration="1m0s" A="b"
+`: `{"caller":"test/output.go:<LINE>","msg":"test","duration":"1h0m0s","X":"y","v":0,"duration":"1m0s","A":"b"}
+`,
+
+		`I output.go:<LINE>] "test" akey9="avalue9" akey8="avalue8" akey1="avalue1" akey5="avalue5" akey4="avalue4"
+`: `{"caller":"test/output.go:<LINE>","msg":"test","akey9":"avalue9","akey8":"avalue8","akey1":"avalue1","v":0,"akey5":"avalue5","akey4":"avalue4"}
+`,
+
+		`I output.go:<LINE>] "test"
+`: `{"caller":"test/output.go:<LINE>","msg":"test","v":0}
+`,
+
+		`I output.go:<LINE>] "\"quoted\"" key="\"quoted value\""
+`: `{"caller":"test/output.go:<LINE>","msg":"\"quoted\"","v":0,"key":"\"quoted value\""}
+`,
+
+		`I output.go:<LINE>] "test" err="whoops"
+`: `{"caller":"test/output.go:<LINE>","msg":"test","v":0,"err":"whoops"}
+`,
+
+		`I output.go:<LINE>] "test" pod="kube-system/pod-1"
+`: `{"caller":"test/output.go:<LINE>","msg":"test","v":0,"pod":{"name":"pod-1","namespace":"kube-system"}}
+`,
+
+		`I output.go:<LINE>] "test" pods=[kube-system/pod-1 kube-system/pod-2]
+`: `{"caller":"test/output.go:<LINE>","msg":"test","v":0,"pods":[{"name":"pod-1","namespace":"kube-system"},{"name":"pod-2","namespace":"kube-system"}]}
+`,
+
+		`I output.go:<LINE>] "test" akey="avalue"
+`: `{"caller":"test/output.go:<LINE>","msg":"test","v":0,"akey":"avalue"}
+`,
+
+		`I output.go:<LINE>] "me: test" akey="avalue"
+`: `{"logger":"me","caller":"test/output.go:<LINE>","msg":"test","v":0,"akey":"avalue"}
+`,
+
+		`I output.go:<LINE>] "test" akey="avalue2"
+`: `{"caller":"test/output.go:<LINE>","msg":"test","akey":"avalue","v":0,"akey":"avalue2"}
+`,
+
+		`I output.go:<LINE>] "you see me"
+`: `{"caller":"test/output.go:<LINE>","msg":"you see me","v":9}
+`,
+
+		`I output.go:<LINE>] "test" firstKey=1
+I output.go:<LINE>] "test" firstKey=1 secondKey=2
+I output.go:<LINE>] "test" firstKey=1
+I output.go:<LINE>] "test" firstKey=1 secondKey=3
+`: `{"caller":"test/output.go:<LINE>","msg":"test","firstKey":1,"v":0}
+{"caller":"test/output.go:<LINE>","msg":"test","firstKey":1,"secondKey":2,"v":0}
+{"caller":"test/output.go:<LINE>","msg":"test","firstKey":1,"v":0}
+{"caller":"test/output.go:<LINE>","msg":"test","firstKey":1,"secondKey":3,"v":0}
+`,
+
+		`I output.go:<LINE>] "odd WithValues" keyWithoutValue="(MISSING)"
+I output.go:<LINE>] "odd WithValues" keyWithoutValue="(MISSING)" anotherKeyWithoutValue="(MISSING)"
+I output.go:<LINE>] "odd WithValues" keyWithoutValue="(MISSING)"
+`: `{"caller":"test/output.go:<WITH-VALUES>","msg":"odd number of arguments passed as key-value pairs for logging","ignored key":"keyWithoutValue"}
```

Comment:
/hold

Let me double-check why these warnings are printed. They should appear when zapr is called indirectly, but not when used directly.

Reply:
/hold cancel